esp/esdllib/esdl_script.cpp

```diff
@@ -1270,6 +1270,66 @@ class CEsdlTransformOperationCopyOf : public CEsdlTransformOperationWithoutChild
     }
 };
 
+class CEsdlTransformOperationTrace : public CEsdlTransformOperationWithoutChildren
+{
+protected:
+    StringAttr m_label;
+    Owned<ICompiledXpath> m_stdout; //optional, if provided use stdout if test evaluates to true
+    Owned<ICompiledXpath> m_test; //optional, if provided trace only if test evaluates to true
+    Owned<ICompiledXpath> m_select;
+
+public:
+    CEsdlTransformOperationTrace(IXmlPullParser &xpp, StartTag &stag, const StringBuffer &prefix) : CEsdlTransformOperationWithoutChildren(xpp, stag, prefix)
+    {
+        m_label.set(stag.getValue("label"));
+        const char *select = stag.getValue("select");
+        if (isEmptyString(select))
+            esdlOperationError(ESDL_SCRIPT_MissingOperationAttr, m_tagname.str(), "without select", m_traceName.str(), !m_ignoreCodingErrors);
+
+        m_select.setown(compileXpath(select));
+
+        const char *test = stag.getValue("test");
+        if (!isEmptyString(test))
+            m_test.setown(compileXpath(test));
+        const char *stdOut = stag.getValue("stdout");
+        if (!isEmptyString(stdOut))
+            m_stdout.setown(compileXpath(stdOut));
```

Comment:
Not sure what the impact of an ESP script requesting stdout could be if the messages are combined with jlog output. This probably shouldn't be a per-request setting, but should be auto-detected (console versus non-console app) or set some other way.